### ISSUE TITLE
Prompt history flickers?

### ISSUE THREAD

**Issue by SirMoogle** [namkazt/sd-webui-prompt-history#42]
I like this extension because it lets me pull up settings for a particular generation, but for some reason the table flickers as if it's constantly refreshing? I tried checking for updates but it seems like I'm already on the latest version.

🤦‍♂️ Just realised this is made for AUTOMATIC1111, so I suspect it might be a compatibility issue.

**Comment by the-real-vortex-v** [comment]
Yeah this was working for me in the past but I'm getting issues in the last few weeks. It looks like it's not caching the thumbnails anymore.

**Comment by malfdawg** [comment]
I'm experiencing the flickering as well (in Forge), that doesn't really bother me as it's merely aesthetic and doesn't affect the functionality, however it is saving my positive prompt twice. Like when I choose a past image, and then apply, the positive prompt is in there double. Negative prompt and adetailer prompts work perfectly. Again, it's not a huge issue as I can just delete the repeated part of the prompt.

**Comment by Hyokkuda** [comment]
I know this topic is old, but this is still an issue today. I looked into the inspect elements and those are the errors:

[76] Attempted to select a non-interactive or hidden tab.

[13] index.js:462 Too many arguments provided for the endpoint.

[Deprecation] Custom state pseudo classes have been changed from ":--color-var" to ":state(color-var)". See more here: https://github.com/w3c/csswg-drafts/issues/4805

The flashing issue seems to be caused by how Forge and the sd-webui-prompt-history extension handle tab visibility and interaction.

I tried to ask around and even got through with Aria AI and such, and that's the only stuff I got. Not enough knowledge about it to fix it myself, unfortunately. I thought increasing the time it takes to refresh the page from 1 second to 60 seconds would help, but it's also affecting the time it takes to load everything the first time. (60 seconds... lol)

" Why This Causes Flashing

- If Forge treats some tabs as "hidden" by default, the function keeps trying to select a hidden tab, leading to a loop of failed selections.
- The function doesn't have a hard stop for cases where a tab isn't interactable, meaning it might keep retrying indefinitely, causing the UI to "flash."
- The issue may be aggravated by Forge’s custom tab handling, which might differ slightly from A1111."

So anyway, I spent an hour or so messing around with the **prompt_history_script.py** from webui\extensions\sd-webui-prompt-history\scripts but, nothing.
